spark/src/main/scala/org/apache/comet/rules/CometScanRule.scala

```diff
@@ -160,42 +151,8 @@ case class CometScanRule(session: SparkSession) extends Rule[SparkPlan] with Com
           scanImpl = selectScan(scanExec, r.partitionSchema, hadoopConf)
         }
 
-        // Native DataFusion doesn't support subqueries/dynamic pruning
-        if (scanImpl == SCAN_NATIVE_DATAFUSION &&
```

Comment:
All of the checks for `native_datafusion` have moved into `CometNativeScan.isSupported(scanExec)`.